### PR TITLE
[metacling] Refactor Property into base class. NFC.

### PR DESCRIPTION
**Pull request by reikdas**
@vgvassilev

### CODE REVIEW

File: core/metacling/src/TClingDeclInfo.h [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #define ROOT_TClingDeclInfo
 
 #include <string>
+#include <clang/AST/Type.h>
```
Comment (vgvassilev) [inline core/metacling/src/TClingDeclInfo.h]: Could you move the clang header before string. We should order includes by generality.

File: core/metacling/src/TClingDeclInfo.h [hunk 2]
```diff
@@ -37,6 +38,7 @@ class TClingDeclInfo {
    }
    virtual bool IsValid() const { return GetDecl(); }
    virtual const char* Name();
+   long Property(long property, clang::QualType qt) const;
```
Comment (vgvassilev) [inline core/metacling/src/TClingDeclInfo.h]: This should be protected, right?

**Comment by vgvassilev** [comment]
@phsft-bot build!

File: core/metacling/src/TClingDeclInfo.h (hunk 1 above)
Comment (reikdas) [reply core/metacling/src/TClingDeclInfo.h]: Fixed.

File: core/metacling/src/TClingDeclInfo.h (hunk 2 above)
Comment (reikdas) [reply core/metacling/src/TClingDeclInfo.h]: Fixed.

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Pull request merged**